py/htm/examples/rest/htm_rest_api.py

```diff
@@ -22,18 +23,12 @@ def request(method, url, data=None, verbose=False):
   if text[0:6] == 'ERROR:':
     raise NetworkRESTError(text[6:].strip())
 
-  if text == 'OK':
+  # determine if result contains JSON i.e. it starts with '{' for a map or '[' for an array
+  if text[0:1] == '[' or text[0:1] == '{':
+    return json.loads( text )
+  else:
     return text
```

Comment:
ok. could even be `try-except` since you return text as fallback anyway (for cases with whitespace, comments etc)

Reply:
> # determine if result contains JSON i.e

I am not happy about this part of the design.  It is still possible that some response that had not intended to be JSON, but pure text, might begin with [ or {.  Leading white space is taken care of because prior to this the value is trimmed. 

Using a try-except would catch most cases where the parser does not recognize a syntax but if there is a real error in the syntax it should throw and not try to use it as text.

 I was thinking that maybe I could rely on the MIME to determine if I should do the JSON decode.  It would depend on the server side knowing what the content contains.

Well, this will do for now but I will think about this.

Reply:
Perhaps we should be consistent and encode ALL response text as JSON, even the OK.  So scalars would be quoted strings.  This is more important now that parameters could be defined as being something more than just a scalar or an array of scalars.  The Value object allows almost any data structure although the getters/setters for parameters do not currently support it.  If we make this all JSON it will allow future changes to the parameter handling.